python/test/unit/language/test_tensor_descriptor.py

```diff
@@ -1434,3 +1436,140 @@ def alloc_fn(size: int, align: int, steam):
 
     ref = torch_scatter_rows(input, idx, y, BLOCK_Y, X, Y)
     torch.testing.assert_close(ref, output, atol=0, rtol=0)
+
+
+NATIVE_SUPPORTED_REDUCE_DTYPES = {
+    "add": {tl.uint32, tl.int32, tl.uint64, tl.float32, tl.float16, tl.bfloat16},
+    "min": {tl.uint32, tl.int32, tl.uint64, tl.int64, tl.float16, tl.bfloat16},
+    "max": {tl.uint32, tl.int32, tl.uint64, tl.int64, tl.float16, tl.bfloat16},
+    "and": {tl.uint32, tl.int32, tl.uint64, tl.int64},
+    "or": {tl.uint32, tl.int32, tl.uint64, tl.int64},
+    "xor": {tl.uint32, tl.int32, tl.uint64, tl.int64},
+}
+FALLBACK_SUPPORTED_REDUCE_DTYPES = {
+    "add": {tl.uint32, tl.int32, tl.uint64, tl.float32, tl.float16, tl.bfloat16},
+    "min": {tl.uint32, tl.int32, tl.uint64, tl.int64},
+    "max": {tl.uint32, tl.int32, tl.uint64, tl.int64},
+    "and": {tl.uint32, tl.int32, tl.uint64, tl.int64},
+    "or": {tl.uint32, tl.int32, tl.uint64, tl.int64},
+    "xor": {tl.uint32, tl.int32, tl.uint64, tl.int64},
+}
+
+
+def min_op(a, b):
+    out = np.minimum(to_numpy(a), to_numpy(b))
+    return unwrap_tensor(to_triton(out, device=a.device))
+
+
+def max_op(a, b):
+    out = np.maximum(to_numpy(a), to_numpy(b))
+    return unwrap_tensor(to_triton(out, device=a.device))
+
+
+REDUCE_OP = {
+    "add": lambda a, b: unwrap_tensor(a) + unwrap_tensor(b),
+    "min": min_op,
+    "max": max_op,
+    "and": lambda a, b: torch.bitwise_and(unwrap_tensor(a), unwrap_tensor(b)),
+    "or": lambda a, b: torch.bitwise_or(unwrap_tensor(a), unwrap_tensor(b)),
+    "xor": lambda a, b: torch.bitwise_xor(unwrap_tensor(a), unwrap_tensor(b)),
+}
+
+REDUCE_SKIP_HIP_CDNA3 = [
+    ("min", "int32", 1, 1024),
+    ("max", "int32", 1, 1024),
+    ("add", "bfloat16", 1, 1024),
+]
+
+
+# TODO: interpreter support
+# @pytest.mark.interpreter
+@pytest.mark.parametrize("kind", ["add", "min", "max", "and", "or", "xor"])
+@pytest.mark.parametrize("dtype_str", tma_dtypes)
+@pytest.mark.parametrize("num_ctas", [1, 2])
+@pytest.mark.parametrize("descriptor", ["host", "device"])
+@pytest.mark.parametrize("M_BLOCK,N_BLOCK", [(2, 16), (8, 16), (8, 32), (8, 128), (512, 32), (1, 1024)])
+def test_tensor_descriptor_reduce(kind, descriptor, dtype_str, num_ctas, M_BLOCK, N_BLOCK):
+    is_native = is_cuda() and torch.cuda.get_device_capability()[0] >= 9
+    if not is_native:
+        if num_ctas != 1:
+            pytest.skip("Multi-CTA not supported")
+        if descriptor == "host":
+            pytest.skip("NYI: Host side tensor descriptor fallback")
+        if is_hip_cdna3() and (kind, dtype_str, M_BLOCK, N_BLOCK) in REDUCE_SKIP_HIP_CDNA3:
+            pytest.skip("Broken on rocm")
```

Comment:
cc @antiagainst I'm getting "failed to translate module to LLVM IR" error but only for specific problem shapes, so seems like an AMD issue.
https://github.com/triton-lang/triton/actions/runs/15045915573/job/42288274328#step:12:3851